bin/detect_reads.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python
+
+import argparse
+import csv
+import sys
+from typing import List, NoReturn
+
+
+def parse_args(args=None) -> argparse.Namespace:
+    """
+    Reformatting is based on detecting whether the reads are paired or single end.
+    Script appends appropriate column to samplesheet.csv file.
+    """
+    Description = "Reformat nf-core/taxprofiler samplesheet file."
+    Epilog = "Example usage: python detect_reads.py <FILE_IN> <FILE_OUT>"
+
+    parser = argparse.ArgumentParser(description=Description, epilog=Epilog)
+    parser.add_argument("FILE_IN", help="Input samplesheet file.")
+    parser.add_argument("FILE_OUT", help="Output file.")
+    return parser.parse_args(args)
+
+
+class ReadsModifier:
+    def __init__(self):
+        self.headers = None
+        self.sample_index = None
+        self.fastq_1_index = None
+        self.fastq_2_index = None
+        self.fasta_index = None
+
+    def detect_reads_and_reformat(self, input_file_path: str, output_file_path: str) -> NoReturn:
+        NEW_COLUMN_NAME = "single_end"
+        new_file_rows = []
+
+        with open(input_file_path, "r") as input_file:
+            csv_reader = csv.reader(input_file, delimiter=",")
+            self.headers = next(csv_reader)
+            self.headers.append(NEW_COLUMN_NAME)
+
+            self._infer_column_indexes()
+
+            for samplesheet_row in csv_reader:
+
+                if self._is_paired_end_short_read(samplesheet_row):
+                    new_file_rows.append([*samplesheet_row, "0"])
```

Comment:
Other tools might parse this as an integer column, maybe single_end true/false is better?

Reply:
This was original behavior. I'm not sure if it won't make taxprofiler to fail